salt/modules/mount.py

```diff
@@ -1422,3 +1569,287 @@ def delete_mount_cache(real_name):
                 if not cache_write:
                     raise CommandExecutionError('Unable to write mount cache.')
     return True
+
+
+def _filesystems(config='/etc/filesystems', leading_key=True):
+    '''
+    .. versionadded:: 2018.3.3
+
+    List the contents of the filesystems
+
+    config
+        File containing filesystem infomation
+
+    leading_key
+        True    return dictionary keyed by 'name' value and dictionary with other keys, values
+                { '/dir' : { 'dev': '/dev/hd8', .... }}
+
+        False   return dictionary  keyed by 'name' value and dictionary with all keys, values
+                { '/dir' : { 'name' : '/dir', 'dev': '/dev/hd8', .... }}
```

Comment:
What is the expected return of this function?

Reply:
OrderedDict, have to maintain the order of the entries found in the file /etc/filesystems as it is significant for correct operation. I shall update the doc string to better reflect this.

Reply:
done